ntt/src/ntt.rs

```diff
@@ -431,21 +432,44 @@ mod tests {
         }
     }
 
-    // TODO Replace by parallel alternative to speed up tests
-    fn transpose<T: Copy>(matrix: &[T], rows: usize, columns: usize) -> Vec<T> {
-        assert_eq!(matrix.len(), rows * columns);
-        let mut v = Vec::with_capacity(matrix.len());
-        for j in 0..columns {
-            for i in 0..rows {
-                v.push(matrix[i * columns + j]);
-            }
-        }
-        v
+    fn transpose<T: Copy + Send + Sync>(matrix: &[T], rows: usize, columns: usize) -> Vec<T> {
+        let len = rows * columns;
+        assert_eq!(matrix.len(), len);
+
+        const TILE_SIZE: usize = 64;
```

Comment:
Tile size should be a function of the cache size and the size of the element. A function like workload_size<F> helps in formulating an equation.